ERCS/erc-7894.md

```diff
@@ -0,0 +1,143 @@
+   - **Function**: vote(uint256 votingIndex_, bool choice_) public
+   - **Description**: Token holders can vote "yes" (true) or "no" (false) on proposals to adjust social points.
+   - **Constraints**:
+     - Caller must own a token.
+     - Each caller can vote only once per proposal.
+
+4. **Ending Voting**
+   - **Function**: endVoting(uint256 votingIndex_) public
+   - **Description**: Ends voting and adjusts points if the majority votes "yes."
+   - **Constraints**:
+     - Voting period must have expired.
+     - Proposal must be active.
+
+5. **Querying Token Points**
+   - **Function**: pointOf(uint256 tokenId) public view returns (uint256)
+   - **Description**: Retrieves the social points associated with a specified token.
+
+6. **Event Triggers**
+   - **NewVotingCreated**: Triggered when a voting proposal is created.
+   - **VoteEvent**: Triggered when a user casts a vote.
+   - **VotingEnd**: Triggered when voting ends.
+   - **PointUpdated**: Triggered when token points are adjusted.
+
+7. **Voting Struct**
+   - **votingIndex**: uint256
+   - **name**: string
+   - **description**: string
+   - **tokenIndex**: uint256
+   - **point**: uint256
+   - **increase**: bool
+   - **yes**: uint256
+   - **no**: uint256
+   - **startDate**: uint256
+   - **endDate**: uint256
+   - **isActive**: bool
+
+## Rationale
+
+SocialSBT decouples governance power from token holdings by implementing a community-based governance system. This ensures fairness and prevents undue influence by high-point token holders. 
+
+Unlike traditional DAOs, where governance power is directly tied to the number of tokens held, SocialSBT enforces a **contribution-based governance model**. However, social points are **not automatically assigned** when a user contributes. Instead, members must perform valuable actions that prompt the community to initiate a voting process to recognize their efforts. This approach ensures that social points are **collectively decided and transparently adjusted** rather than arbitrarily awarded.  
+
+Here’s how SocialSBT mitigates whale attacks in practice:
+
+1. **Non-transferable tokens**: Since SocialSBT tokens cannot be bought or transferred, governance power cannot be acquired through market purchases.
+2. **Community-driven social point adjustments**: Social points are not automatically granted. Instead, a proposal must be initiated and approved by the DAO community for a member’s points to increase or decrease.
+3. **Contribution-based voting power**: Only members who provide meaningful contributions can gain governance influence, and these contributions must be recognized by the community through an **on-chain voting process**.
+
+Because SocialSBT ensures that governance influence is directly linked to community engagement rather than wealth, whales cannot manipulate voting outcomes by simply acquiring tokens. This system reinforces democratic participation and aligns decision-making with the collective interests of the DAO.
```

Comment:
Most of this belongs in Motivation. The Rationale section should explain technical decisions made while writing the proposal, where the Motivation section justifies the proposal as a whole. My favourite analogy is:

> Motivation: We need to build a shed because...
> Rationale: We chose to paint the shed red because...